scripts/dts/extract/globals.py

```diff
@@ -351,38 +351,63 @@ def get_binding(node_address):
 def get_binding_compats():
     return bindings_compat
 
+def build_cell_array(prop_array):
+
+    index = 0
+    ret_array = []
+
+    while index < len(prop_array):
+        handle = prop_array[index]
+
+        if handle in {0, -1}:
+            ret_array.append([])
+            index += 1
+        else:
+            # get controller node (referenced via phandle)
+            cell_parent = phandles[handle]
+
+            for k in reduced[cell_parent]['props'].keys():
+                if k[0] == '#' and '-cells' in k:
+                    num_cells = reduced[cell_parent]['props'].get(k)
+
+            ret_array.append(prop_array[index:index+num_cells+1])
+
+            index += num_cells + 1
+
+    return ret_array
+
+
 def extract_controller(node_address, prop, prop_values, index,
                        def_label, generic, handle_single=False):
 
     prop_def = {}
     prop_alias = {}
 
-    # get controller node (referenced via phandle)
-    cell_parent = phandles[prop_values[0]]
+    p = build_cell_array(prop_values)
```

Comment:
What does `p` stand for? Could use a more descriptive name maybe.

Reply:
Done